features/cellular/framework/AT/ATHandler.cpp

```diff
@@ -295,12 +310,15 @@ void ATHandler::restore_at_timeout()
 
 void ATHandler::process_oob()
 {
+    ScopedLock<ATHandler> lock(*this);
+    _oob_pending = true;
+    _oob_queued = false;
     if (!_is_fh_usable) {
         tr_debug("process_oob, filehandle is not usable, return...");
+        _oob_pending = false;
+        _oobCv.notify_all();
```

Comment:
This would need to be conditional on `AT_HANDLER_MUTEX`

Reply:
added #ifdef